marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/cloud/azure/AzureKeyVaultSecret.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2012-2021 MarkLogic Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+package com.marklogic.hub.central.cloud.azure;
+
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import com.azure.security.keyvault.secrets.SecretClient;
+import com.azure.security.keyvault.secrets.SecretClientBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+@Profile("azure")
+public class AzureKeyVaultSecret {
```

Comment:
I think "AzureKeyVault" is sufficient for a name; if "Secret" were used, it seems it should really be "SecretClient", but I don't think that's necessary to add here. "AzureKeyVault" implies what this class represents, which is an interface for getting parameter values from an Azure key vault.

Reply:
Azure KeyVault service can be used to store keys, secrets, and certificates. This class will only retrieve a "secret" hence the name.